src/Stratis.Bitcoin/P2P/PeerConnector.cs

```diff
@@ -191,27 +184,22 @@ public void StartConnectAsync()
 
             this.asyncLoop = this.asyncLoopFactory.Run($"{this.GetType().Name}.{nameof(this.ConnectAsync)}", async token =>
             {
-                await this.ConnectAsync();
+                if (!this.peerAddressManager.Peers.Any())
+                    return;
+
+                if (this.ConnectedPeers.Count >= this.MaximumNodeConnections)
+                    return;
+
+                await this.OnConnectAsync();
             },
             this.nodeLifetime.ApplicationStopping,
             repeatEvery: TimeSpans.Second);
         }
 
         /// <summary>Attempts to connect to a random peer.</summary>
-        private async Task ConnectAsync()
+        internal async Task ConnectAsync(PeerAddress peerAddress)
         {
-            if (!this.peerAddressManager.Peers.Any())
-                return;
-
-            if (this.ConnectedPeers.Count >= this.MaximumNodeConnections)
-                return;
-
-            PeerAddress peerAddress = this.FindPeerToConnectTo();
-            if (peerAddress == null)
-            {
-                Task.Delay(TimeSpans.TenSeconds.Milliseconds).Wait(this.nodeLifetime.ApplicationStopping);
-                return;
-            }
+            this.logger.LogTrace("()");
```

Comment:
you are missing matching exit log
and you should also log the argument

Reply:
Fixed